n64.ld

```diff
@@ -20,6 +20,8 @@
 OUTPUT_FORMAT ("elf32-bigmips", "elf32-bigmips", "elf32-littlemips")
 OUTPUT_ARCH (mips)
 EXTERN (_start)
+EXTERN(__init_interrupts)
+EXTERN(__init_cop1)
```

Comment:
I think the idea here is that the compiler will auto-remove constructors of modules that are not used by the application. This is actually a good feature for us, since we want in general to reduce code size.

In this specific case, I would say that the interrupt constructor doesn't need extern: if I manage to do a main that never calls disable/enable interrupts (!), then I would not need init_interrupts as well. The interrupts would stay disabled and that's it.

On the other, I think the issue is that you put the constructor in cop1.c which is a new object file and otherwise unreferenced. If you moved that to something like n64sys.c, there wouldn't be a need for this.

What do you think?

Reply:
I like the idea of not having a new file for this one routine. I'll move it into n64sys.c, I think the EXTERN will still be required to make sure the cop1 bits are set, as I don't think every program is guaranteed to use a symbol from n64sys.c

Reply:
I think n64sys is always included because it contains get_memory_size which is used by sbrk. But I see the point of this being a little fragile in general, so the EXTERN is probably worth it anyway.